Frameworks/UIKit.Xaml/TextField.xaml.cpp

```diff
@@ -0,0 +1,759 @@
+    } else {
+        return TextBox->Focus(FocusState::Programmatic);
+    }
+}
+
+void TextField::OnTextChanged(Platform::Object ^sender, Windows::UI::Xaml::Controls::TextChangedEventArgs ^e)
+{
+    if (_textChangedHandler) {
+        _textChangedHandler->Invoke(
+            InspectableFromObject(this).Get(),
+            ObjectToType<ABI::Windows::UI::Xaml::IRoutedEventArgs>(e).Get());
+    }
+}
+
+void TextField::OnPasswordChanged(Platform::Object ^sender, Windows::UI::Xaml::RoutedEventArgs ^e)
+{
+    if (_textChangedHandler) {
+        _textChangedHandler->Invoke(
+            InspectableFromObject(this).Get(),
+            ObjectToType<ABI::Windows::UI::Xaml::IRoutedEventArgs>(e).Get());
+    }
+}
+
+void TextField::OnGotFocus(Platform::Object ^sender, Windows::UI::Xaml::RoutedEventArgs ^e)
+{
+    if (_gotFocusHandler) {
+        _gotFocusHandler->Invoke(
+            InspectableFromObject(this).Get(),
+            ObjectToType<ABI::Windows::UI::Xaml::IRoutedEventArgs>(e).Get());
+    }
+}
+
+void TextField::OnLostFocus(Platform::Object ^sender, Windows::UI::Xaml::RoutedEventArgs ^e)
+{
+    if (_lostFocusHanlder) {
+        _lostFocusHanlder->Invoke(
+            InspectableFromObject(this).Get(),
+            ObjectToType<ABI::Windows::UI::Xaml::IRoutedEventArgs>(e).Get());
+    }
+}
+
+void TextField::OnKeyDown(Platform::Object ^sender, Windows::UI::Xaml::Input::KeyRoutedEventArgs ^e)
+{
+    if (_enterKeyDownHandler) {
+        if (e->Key == Windows::System::VirtualKey::Enter) {
+            _enterKeyDownHandler->Invoke(
+                InspectableFromObject(this).Get(),
+                ObjectToType<ABI::Windows::UI::Xaml::IRoutedEventArgs>(e).Get());
+            e->Handled = true;
+        } else {
+            e->Handled = false;
+        }
+    }
+}
+
+void TextField::ApplyFont(Platform::String^ fontFamily, FontStretch strech, FontStyle fontStyle, double fontSize, FontWeight fontWeight) {
+    if (SecureEntry) {
+        PasswordBox->FontFamily = ref new FontFamily(fontFamily);
+        PasswordBox->FontStretch = strech;
+        PasswordBox->FontStyle = fontStyle;
+        PasswordBox->FontSize = fontSize;
+        PasswordBox->FontWeight = fontWeight;
+    }
+    else {
+        TextBox->FontFamily = ref new FontFamily(fontFamily);
+        TextBox->FontStretch = strech;
+        TextBox->FontStyle = fontStyle;
+        TextBox->FontSize = fontSize;
+        TextBox->FontWeight = fontWeight;
+    }
+}
+
+void TextField::_SetTextVerticalAlignment() {
+    // Set vertical aligment of both content and placeholder to the same value
+    FrameworkElement^ target = FindTemplateChild<FrameworkElement^>(_backingControl, "ContentElement");
+    if (target) {
+        target->VerticalAlignment = VerticalTextAlignment;
+    }
+
+    target = FindTemplateChild<FrameworkElement^>(_backingControl, "PlaceholderTextContentPresenter");;
+    if (target) {
+        target->VerticalAlignment = VerticalTextAlignment;
+    }
```

Comment:
does this work for both TextBox and PasswordBox?  Might want to add a few more comments about how you know where to find these, since the Xaml markup is practically empty.